src/query/execute_schema.c

```diff
@@ -8818,6 +8865,31 @@ do_create_entity (PARSER_CONTEXT * parser, PT_NODE * node)
 	      do_flush_class_mop = true;
 	    }
 	}
+      if (tbl_opt_encrypt)
+	{
+	  encrypt_node = tbl_opt_encrypt->info.table_option.val;
+	  assert (encrypt_node != NULL && encrypt_node->node_type == PT_VALUE
+		  && encrypt_node->type_enum == PT_TYPE_INTEGER);
+	  tde_algo_opt = encrypt_node->info.value.data_value.i;
+	  /*
+	   *  -1 means using deafult encryption algorithm.
+	   *  Other values but -1, TDE_ALGORITHM_AES, TDE_ALGORITHM_ARIA has been denied by parser.
+	   */
+	  if (tde_algo_opt == -1)
+	    {
+	      tde_algo = (TDE_ALGORITHM) prm_get_integer_value (PRM_ID_TDE_DEFAULT_ALGORITHM);
+	    }
+	  else
+	    {
+	      tde_algo = (TDE_ALGORITHM) tde_algo_opt;
+	    }
+	  error = sm_set_class_tde_algorithm (class_obj, tde_algo);
+	  if (error != NO_ERROR)
+	    {
+	      break;
+	    }
+	  do_flush_class_mop = true;
```

Comment:
returned error can be overwritten by followed codes.

Reply:
yes is there is no error

Reply:
Ok, I confused.